internal/server/data/group.go

```diff
@@ -41,19 +40,37 @@ func ListGroups(db GormTxn, p *Pagination, selectors ...SelectorFunc) ([]models.
 	}
 
 	return groups, nil
-
 }
 
 func ByGroupMember(id uid.ID) SelectorFunc {
 	return func(db *gorm.DB) *gorm.DB {
 		return db.
-			Joins("JOIN identities_groups ON groups.id = identities_groups.group_id").
-			Where("identities_groups.identity_id = ?", id)
+			Joins(`
+			JOIN (
+				-- Combine the direct group members and the members of mapped identity provider groups
+				SELECT * FROM identities_groups
+				NATURAL FULL JOIN (
+					-- See identities in groups based on their provider group mappings
+					SELECT identity_id, group_id from provider_groups_identities
+					JOIN provider_groups_groups ON provider_groups_groups.provider_group_id = provider_groups_identities.provider_group_id
+				) AS normalized_identity_groups
+				) AS resolved_identity_groups
+			ON resolved_identity_groups.group_id = id
+			`).
+			Where("resolved_identity_groups.identity_id = ?", id)
 	}
 }
 
 func groupIDsForUser(tx ReadTxn, userID uid.ID) ([]uid.ID, error) {
-	stmt := `SELECT DISTINCT group_id FROM identities_groups WHERE identity_id = ?`
+	stmt := `
+		SELECT DISTINCT group_id FROM identities_groups
+		NATURAL FULL JOIN (
+			-- See identities in groups based on their provider group mappings
+			SELECT identity_id, group_id from provider_groups_identities
+			JOIN provider_groups_groups ON provider_groups_groups.provider_group_id = provider_groups_identities.provider_group_id
+		) AS normalized_identity_groups
+		WHERE identity_id = ?
+	`
```

Comment:
also this should not be querying provider_groups_identities. I see what you're going for here, that the group membership is the set of all manually added users plus all users added from the provider_group, but I think it's much easier (and much more performant) to simply update groups_identities when the provider_group says it should change. This keeps the complication from bleeding into our app.